dbbuild/lib/marker.py

```diff
@@ -168,6 +168,9 @@ def chrom_num(self):
     def locus(self):
         return self.name.split(".")[0]
 
+    def overlaps(self, other):
+        return self.start <= other.end and self.end >= other.start
+
```

Comment:
Shouldn't there also be a check to make sure they are on the same chromosome? Also if my interval math is correct ( which is always a big if 😅 ), for half-open intervals the overlap calculation would be `self.start < other.end and self.end > other.start`

Reply:
> Shouldn't there also be a check to make sure they are on the same chromosome?

...yes

I'll take care of that right away! 😀

Reply:
> ...for half-open intervals...

These are actually 1-based inclusive intervals. The amount of interval arithmetic performed in MicroHapDB is pretty limited, so while I usually like to use 0-based half-open intervals in code, I haven't taken the time to switch over in MicroHapDB.

Reply:
Actually, the best check would be to make sure that `microhapdb marker --format=fasta` doesn't create any ridiculously long sequences, since that is what caught this bug in the first place. Probably should add a test for that...